### DISCUSSION TITLE
Adjusting Layout Behavior When Collapsing FluentNavMenu in Example/Demo Project

### DISCUSSION THREAD

**Discussion by thinkzeno** [microsoft/fluentui-blazor#1303] [Q&A] [answered]
When the 'Collapsible' property of the 'FluentNavMenu' used in the 'Example / Demo Project' layout is set to 'true,' it allows toggling between collapse and expand. However, when the menu narrows down to 40px in width, the main body area does not shift to the left and remains unchanged. How can I modify this behavior?

**examples > Demo > Shared > Shared > DemoNavMenu.razor**
<div class="navmenu">
    <input type="checkbox" title="Menu expand/collapse toggle" id="navmenu-toggle" class="navmenu-icon" />
    <label for="navmenu-toggle" class="navmenu-icon"><FluentIcon Value="@(new Icons.Regular.Size20.Navigation())" Color="Color.Neutral" /></label>
    <nav class="sitenav" aria-labelledby="main-menu" onclick="document.getElementById('navmenu-toggle').click()">
        <FluentNavMenu Id="main-menu" _**Collapsible="true"**_ Title="Main menu">
            <FluentNavLink Icon="@(new Icons.Regular.Size20.Home())" Href="/" Match="NavLinkMatch.All">
                <h3>Home</h3>
            </FluentNavLink>

![이미지 2024-01-11 오후 9 25](https://github.com/microsoft/fluentui-blazor/assets/61155281/4aa9e378-f0c6-4a78-865c-cba00d950707)

**Comment by vnbaaij** [comment]
Hi,

You can see how that is done in the sites that get created when using our templates as a starting point. See https://www.fluentui-blazor.net/Templates on how to set that up.

**Reply by thinkzeno** [reply]
Thank you for your response. In my view, there are two issues with the template. The footer doesn't consistently stay fixed, and the scrollbar behaviors differ from the example. I appreciate the layout implementation in the example, except for the aspects I mentioned.

**Reply by vnbaaij** [reply, marked as answer]
I'm making a change to the height of  `.main` element for next release so footer stays fixed.
Not sure what you mean with the 'scrollbar behaviors`.

**Reply by thinkzeno** [reply]
I would like to suggest that, similar to the example, having a scrollbar on the right side of the 'NavBar' seems more natural and visually appealing than the scrollbar on the right side of the page.

**Reply by vnbaaij** [reply]
We are not going to change that behavior because we don't want to deviate from the standard templates too much

**Reply by gligorov** [reply]
> We are not going to change that behavior because we don't want to deviate from the standard templates too much

Hello, @vnbaaij .   Is there any update regarding this issue.  I have the same problem and feel that when the menu is collapsed, the page content should expand to use all the space.  It feels strange like this no.  The image on this issue describes exactly the problem.

I have a latest current Blazor Server app, used the VS Template project and .NET 9 just made the app a week ago, and trying to see if there is any update on this issue.  Thanks.

**Reply by vnbaaij** [reply]
@gligorov you are talking about a different issue then what my remark was about.

When using our templates, I believe the main area is expanded if you collapse the nav menu. If not, it can be changed to do that pretty easily with some standard HTML and CSS. See https://fluentui-blazor.net/NavMenu#collapsible Navigation for an example of that

**Reply by gligorov** [reply]
Thank you @vnbaaij, understand.  I just made a new project using VS Wizzard and few changes, and now the expansion works perfectly on Windows under Edge.  It does not work on mobile device for some reason.  See my screen, this is same app hosted under IIS online in my dev server.

![image](https://github.com/user-attachments/assets/9e6cc391-7119-41ed-8302-48e1fbb7a292)

Mobile view not good, from my Samsung S24 Ultra.
![image](https://github.com/user-attachments/assets/31a23143-247e-42f9-8f11-29cc0ddccca7)

**Reply by vnbaaij** [reply]
Probably a CSS issue in a media query

**Reply by gligorov** [reply]
k, let me check to see what could be causing it, thank you!

**Reply by gligorov** [reply]
Yes that was it, there ware couple of specs below in the mobile media query,  removing those solved the problem.

  #main-menu {   width: 100% !important;   }
    .navmenu {  width: 100%;  }
    
   Thanks again, for the quick pointers.
